PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/core/AnnotationHelper.java

```diff
@@ -0,0 +1,114 @@
+import com.intellij.psi.PsiReferenceExpression;
+import com.intellij.psi.impl.JavaConstantExpressionEvaluator;
+import com.microsoft.azure.common.exceptions.AzureExecutionException;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
+
+public class AnnotationHelper {
+
+    public static String getJvmAnnotationAttributeValue(JvmAnnotationAttributeValue value) throws AzureExecutionException {
+        if (value instanceof JvmAnnotationConstantValue) {
+            return Objects.toString(((JvmAnnotationConstantValue) value).getConstantValue(), null);
+        }
+        if (value instanceof PsiExpression) {
+            if (value instanceof PsiReferenceExpression) {
+                final PsiReferenceExpression referenceExpression = (PsiReferenceExpression) value;
+                final Object resolved = referenceExpression.resolve();
+                if (resolved instanceof PsiEnumConstant) {
+                    final PsiEnumConstant enumConstant = (PsiEnumConstant) resolved;
+                    final PsiClass enumClass = enumConstant.getContainingClass();
+                    if (enumClass != null) {
+                        try {
+                            return getEnumFieldString(enumClass.getQualifiedName(), enumConstant.getName());
+                        } catch (ClassNotFoundException | IllegalAccessException e) {
+                            throw new AzureExecutionException(e.getMessage(), e);
+                        }
+                    } else {
+                        return enumConstant.getName();
+                    }
+                }
+
+            }
+            final Object obj = JavaConstantExpressionEvaluator.computeConstantExpression((PsiExpression) value, true);
+            return Objects.toString(obj, null);
+        } else if (value instanceof PsiLiteral) {
+            return Objects.toString(((PsiLiteral) value).getValue(), null);
+        }
+        throw new AzureExecutionException("Cannot get annotation value of type : " + value.getClass().getName());
+    }
+
+    public static String getPsiAnnotationMemberValue(PsiAnnotationMemberValue value) throws AzureExecutionException {
+        if (value instanceof PsiExpression) {
+            if (value instanceof PsiReferenceExpression) {
+                final PsiReferenceExpression referenceExpression = (PsiReferenceExpression) value;
+                final Object resolved = referenceExpression.resolve();
+                if (resolved instanceof PsiEnumConstant) {
+                    final PsiEnumConstant enumConstant = (PsiEnumConstant) resolved;
+                    final PsiClass enumClass = enumConstant.getContainingClass();
+                    if (enumClass != null) {
+                        try {
+                            return getEnumFieldString(enumClass.getQualifiedName(), enumConstant.getName());
+                        } catch (ClassNotFoundException e) {
```

Comment:
handle the exception with AzureExecutionException

Reply:
Done